common/persistence/dataInterfaces.go

```diff
@@ -1424,6 +1424,30 @@ type (
 		Branches []HistoryBranchDetail
 	}
 
+	// ScanWorkflowsRequest is a request of ScanCurrentWorkflows
+	ScanWorkflowsRequest struct {
+		// pagination token
+		NextPageToken []byte
+		PageSize int
+		ShardID  int
+	}
+
+	// ScanWorkflowsResponse is a response to ScanWorkflows
+	ScanWorkflowsResponse struct {
+		// pagination token
+		NextPageToken []byte
+		WorkflowExecutions []ScanWorkflowDetailInfo
+	}
+
+	// ScanWorkflowDetailInfo is detail of ScanWorkflowsResponse
+	ScanWorkflowDetailInfo struct{
+		DomainID string
+		WorkflowID string
+		RunID  string
+		State  int
+		IsCurrentWorkflow bool
```

Comment:
this seems very cassandra specific i.e. in cassandra, both mutable state and current_record are stored in the executions table. But this will not be the case in SQL based stores. In mySQL for example, we have separate tables for execution and current_execution. Have the contract for this API return results from both is a bit confusing as well as will complicate pageToken implementation for other stores. We have two options:

 - Create separate APIs
 -  Add request filters to indicate what type of results is desired

Reply:
Let's chat about this one more. The hard part here is this API is meant to abstract out the underlying storage technology, but by virtue of different storage technologies potentially using different schemas it means the response is not uniform across technologies.